mirdata/irmas.py

```diff
@@ -112,69 +112,31 @@
-                                            'genre': genre_code,
-                                            'drum': False,
-                                            'train': True,
-                                        }
-                                    else:
-                                        irmas_id = file.split(']')[2]  # Obtain id
-                                        irmas_id_nod_no_wav = irmas_id.split('.')[
-                                            0
-                                        ]  # Obtain id without '.wav'
-                                        split_1 = file.split('[')[2]
-                                        genre_code = split_1.split(']')[0]
-                                        irmas_dict[irmas_id_nod_no_wav] = {
-                                            'genre': genre_code,
-                                            'drum': None,
-                                            'train': True,
-                                        }
-            if 'Test' in directory:
-                for root_, dirs_, files_ in os.walk(os.path.join(data_home, directory)):
-                    for directory_ in dirs_:
-                        for root__, dirs__, files__ in os.walk(
-                            os.path.join(data_home, directory, directory_)
-                        ):
-                            for file in files__:
-                                if file.endswith('.wav'):
-                                    test_index = str(count)
-                                    irmas_dict[test_index] = {
-                                        'genre': None,
-                                        'drum': None,
-                                        'train': False,
-                                    }
-                                    count += 1
+    with open('mirdata/indexes/irmas_index.json') as json_file:
+        index = json.load(json_file)
+        for track_id in index.keys():
+            if '__' in track_id:
+                if 'dru' in index[track_id]['audio'][0] or 'nod' in index[track_id]['audio'][0]:
+                    genre = index[track_id]['audio'][0].split('.')[0].split('[')[3].split(']')[0]
+                    irmas_dict[track_id] = {
+                        'genre': genre,
+                        'drum': [True if 'dru' in index[track_id]['audio'][0] else False][0],
+                        'train': True
+                    }
+                else:
+                    genre = index[track_id]['audio'][0].split('.')[0].split('[')[2].split(']')[0]
+                    irmas_dict[track_id] = {
+                        'genre': genre,
+                        'drum': None,
+                        'train': True
+                    }
+            else:
+                irmas_dict[track_id] = {
+                    'genre': None,
+                    'drum': None,
+                    'train': False
+                }
```

Comment:
This is looking much better! I think I understand better the structure as well.

Ideally we can use `DATA = utils.LargeData("tinysol_index.json", _load_metadata)` which takes care of loading both the index and the metadata (and they are accessed as `DATA.index` and `DATA.metadata`, see [here for an example](https://github.com/mir-dataset-loaders/mirdata/blob/master/mirdata/tinysol.py)). However this dataset seems tricky since you have to parse the wav files. Now that the code is simplified let's see if we can change things to use the `LargeData` object.  Can you explain what's the directory structure a bit?

Reply:
So you would have a general IRMAS folder. Into the IRMAS folder there are 4 folders:

- IRMAS-TrainingData
Inside this folder you can find several folders, each of them referring to a different instrument. Exemples: cla/, flu/, etc. The samples inside flu/ will be tracks were a flute is the predominant instrument. This track have all the annotations in the filename: predominant instrument, genre and drum presence (sometimes), plus an unique ID.


- IRMAS-TestingData-Part1/Part1 (inside the TestingData folders there is a README file with the information written in the documentation and then a folder named PartX with the files)
- IRMAS-TestingData-Part2/Part2
- IRMAS-TestingData-Part2/Part3
Inside IRMAS-TestingData-PartX/PartX you can find the testing samples belonging to that part of the testing data. They are random song excerpts from different musical styles and origins. Every .wav track is accompanied with a .txt file (that has the same name) with the annotation of the predominant instrument(s). Example: Highway_to_hell.wav and Highway_to_hell.txt, and inside the .txt file you would find "voi /n gel" (voi refers to voice and gel to electric guitar).

Here you can find the [Zenodo link to the dataset](https://zenodo.org/record/1290750#.X47x51ngq8U) in case my explanation is confusing or not enough clear:)

Reply:
@rabitt, apparently to get the metadata for this dataset as we usually do we should parse a bunch of files down in several folders as @genisplaja was doing, but is a lot of code and walk through directories. Instead we're parsing the index, but that goes against what we usually do that is to use the `LargeData` object: in this case we're parsing the index in `_load_metadata` **and** using the `DATA` object. What are your thoughts on this now that you're doing the `Dataset` object?

Reply:
I think this dataset doesn't actually need a `_load_metadata` function at all. Lines 118-138 can be moved almost exactly into the `__init__` of the `Track` object, replacing lines 182-193! since everything just depends on the index and not on any external metadata file, you have all the information you need directly inside the Track object (via `self._track_paths`). It will make the init a bit more verbose, but its ok!

Reply:
Ok! Fixed now! Sorry for the delay.